pysteps/postprocessing/probmatching.py

```diff
@@ -259,6 +260,57 @@ def _get_error(scale):
     return shift, scale, R.reshape(shape)
 
 
+def resample_distributions(first_array, second_array, probability_first_array):
+    """
+    Merges two distributions (e.g. from the extrapolation nowcast and NWP in the blending module)
+    to effectively combine two distributions for the probability matching without losing extremes.
+    Parameters
+    ----------
+    first_array: array_like
+        One of the two arrays from which the distribution should be samples (e.g. the extrapolation
+        cascade).
+    second_array: array_like
+        One of the two arrays from which the distribution should be samples (e.g. the NWP (model)
+        cascade).
+    probability_first_array: float
+        The weight that first_array should get (as a value between 0 and 1). This is
+        typically based on cascade level 2 (as is done for all the probability matching steps in `blending/steps.py`)
+
+    Returns
+    ----------
+    csort: array_like
+        The output distribution as first_array drawn binomial distribution from the input arrays first_array and second_array.
+    """
+    # Make sure that the probability is never: <0, >1 or nan
+    if probability_first_array < 0.0:
+        probability_first_array = 0.0
+    elif probability_first_array > 1.0:
+        probability_first_array = 1.0
+    elif np.isnan(probability_first_array):
+        probability_first_array = 0.0
+
+    # First make sure there are no nans in the input data
+    # Where the extrapolation cascade is nan (outside the radar domain), we fill it up with the model data
```

Comment:
here to, I think this should be handled in `blending/steps.py`

Reply:
Is this part necessary @RubenImhoff ?

Reply:
Not necessarily this way, but we should account for nans in the 2D input arrays. My suggestion would be that if there is a nan at a location in one of the two input arrays, you automatically sample the non-nan value in the array that contains a value at that cell. A cleaner way to account for this would be good, I think.

Reply:
OK I see, a bit the problem is that this function is global in that it doesn't compare values at locations, but only considers their ranks. (that is, the arrays are sorted before being resampled). the  logic you describe is in my opinion something different that is specific to steps blending and therefore it should be done before calling this function. Or am I misunderstanding something?

Reply:
That is certainly true, although nans will almost always occur with the datasets that are used for pysteps. So, the question is if we would like to account for it, or if we just want to leave this up to the user and thus put it in the `blending/steps.py` code (fine with me!).

Reply:
OK please have a loot at [a46e62a](https://github.com/pySTEPS/pysteps/pull/390/commits/a46e62a58aa0af64e7421bc8d273f0696974f389) and let me know if this works for you

Reply:
Great, thanks, Daniele! This looks good to me.